server/src/main/java/org/opensearch/index/remote/RemoteRefreshSegmentTracker.java

```diff
@@ -361,12 +369,36 @@ long getRejectionCount(String rejectionReason) {
         return rejectionCountMap.get(rejectionReason).get();
     }
 
-    Map<String, Long> getLatestLocalFileNameLengthMap() {
-        return latestLocalFileNameLengthMap;
+    public Map<String, Long> getLatestLocalFileNameLengthMap() {
+        return Collections.unmodifiableMap(latestLocalFileNameLengthMap);
     }
 
-    public void setLatestLocalFileNameLengthMap(Map<String, Long> latestLocalFileNameLengthMap) {
-        this.latestLocalFileNameLengthMap = latestLocalFileNameLengthMap;
+    /**
+     * Updates the latestLocalFileNameLengthMap by adding file name and it's size to the map. The method is given a function as an argument which is used for determining the file size (length in bytes). This method is also provided the collection of segment files which are the latest refresh local segment files. This method also removes the stale segment files from the map that are not part of the input segment files.
+     *
+     * @param segmentFiles list of local refreshed segment files
+     * @param fileSizeFunction function is used to determine the file size in bytes
+     */
+    public void updateLatestLocalFileNameLengthMap(
+        Collection<String> segmentFiles,
+        CheckedFunction<String, Long, IOException> fileSizeFunction
+    ) {
+        // Update the map
+        segmentFiles.stream()
+            .filter(file -> EXCLUDE_FILES.contains(file) == false)
+            .filter(file -> latestLocalFileNameLengthMap.containsKey(file) == false || latestLocalFileNameLengthMap.get(file) == 0)
+            .forEach(file -> {
+                long fileSize = 0;
+                try {
+                    fileSize = fileSizeFunction.apply(file);
+                } catch (IOException e) {
+                    logger.warn(new ParameterizedMessage("Exception while reading the fileLength of file={}", file), e);
+                }
+                latestLocalFileNameLengthMap.put(file, fileSize);
+            });
+        Set<String> fileSet = new HashSet<>(segmentFiles);
+        // Remove keys from the fileSizeMap that do not exist in the latest segment files
+        latestLocalFileNameLengthMap.entrySet().removeIf(entry -> fileSet.contains(entry.getKey()) == false);
```

Comment:
Is it safe to assume that `EXCLUDE_FILES` will never be contained in `latestLocalFileNameLengthMap`? The above block excludes these, but are there other places that need to be checked? 
Asking because this line would skip removing `EXCLUDE_FILES` if they are contained in both `latestLocalFileNameLengthMap` and `fileSet`.

Reply:
> Is it safe to assume that `EXCLUDE_FILES` will never be contained in `latestLocalFileNameLengthMap`? The above block excludes these, but are there other places that need to be checked?

Yes. It would not be contained as there is a filter that is excluding these files. We are also skipping this during the upload.

> Asking because this line would skip removing `EXCLUDE_FILES` if they are contained in both `latestLocalFileNameLengthMap` and fileSet.

We would not putting the file in map if it is present in `EXCLUDE_FILES`. https://github.com/opensearch-project/OpenSearch/pull/8758/files#diff-37a64296971dc62e523f4abab8db543f5c11f4c49e3edef40609701ff469244aR401 -> this is for cleaning up the files that are no more part of the latest refresh segment files.